code/datums/crew_credits.dm

```diff
@@ -0,0 +1,276 @@
+/datum/crewCredits/proc/bundle_crew_member_data(var/datum/mind/M)
+	var/is_head = false
+	if(M.special_role)
+		if(!M.current) return
+
+		. += list(list(
+			"real_name" = M.current.real_name,
+			"dead" = isdead(M.current),
+			"player" = M.displayed_key,
+			"role" = M.special_role,
+			"head" = is_head,
+		))
+		return .
+
+	if(!M.assigned_role) return
+
+	if (M.assigned_role in list("Head of Security", "Head of Personnel", "Medical Director", "Research Director", "Chief Engineer", "AI"))
+		is_head = true
+
+	. +=list(list(
+		"real_name" = M.current.real_name,
+		"dead" = isdead(M.current),
+		"player" = M.displayed_key,
+		"role" = M.assigned_role,
+		"head" = is_head,
+	))
+
+/datum/crewCredits/ui_static_data(mob/user)
+	. = ..()
+	if(crew_creds)
+		logTheThing(LOG_DEBUG, null, "Zamujasa/CREWCREDITS: [world.timeofday] returning already-generated crew credits")
+		return crew_creds
+
+	logTheThing(LOG_DEBUG, null, "Zamujasa/CREWCREDITS: [world.timeofday] starting crew credits generation")
+	var/list/antagonist = list()
+	var/list/captain = list()
+	var/list/security = list()
+	var/list/medical = list()
+	var/list/science = list()
+	var/list/engineering = list()
+	var/list/civilian = list()
+	var/list/silicon = list()
+	var/list/other = list()
+
+	for(var/datum/mind/M in ticker.minds)
+
+		// Antagonist?
+		if(M.special_role)
+			antagonist += bundle_crew_member_data(M)
+		if(!M.assigned_role)
+			continue
+
+		if (M.assigned_role == "Captain")
+			captain += bundle_crew_member_data(M)
+		else if ((M.assigned_role in security_jobs) || (M.assigned_role in security_gimmicks))
+			security += bundle_crew_member_data(M)
+		else if ((M.assigned_role in medical_jobs) || (M.assigned_role in medical_gimmicks))
+			medical += bundle_crew_member_data(M)
+		else if ((M.assigned_role in science_jobs) || (M.assigned_role in science_gimmicks))
+			science += bundle_crew_member_data(M)
+		else if ((M.assigned_role in engineering_jobs) || (M.assigned_role in engineering_gimmicks))
+			engineering += bundle_crew_member_data(M)
+		else if ((M.assigned_role in service_jobs) || (M.assigned_role in service_gimmicks) || M.assigned_role == "Staff Assistant")
+			civilian += bundle_crew_member_data(M)
+		else if ((M.assigned_role == "AI") || (M.assigned_role == "Cyborg"))
+			silicon += bundle_crew_member_data(M)
+		else if (M.assigned_role == "Pathologist")
+			#ifdef SCIENCE_PATHO_MAP
+			science += bundle_crew_member_data(M)
+			#else
+			medical += bundle_crew_member_data(M)
+			#endif
+		else
+			other += bundle_crew_member_data(M)
+
+	logTheThing(LOG_DEBUG, null, "Zamujasa/CREWCREDITS: [world.timeofday] done processing minds. info: A [antagonist.len] C [captain.len] S [security.len] M [medical.len] R [science.len] E [engineering.len] Cv [civilian.len] X [other.len]")
+
+	/* ~ BEGIN FAKE CREW CREDITS GENERATION ~ */
+	#ifdef CREDITS_DEBUGGING
+
+	var/has_head = FALSE
+	var/what_role = ""
+
+	while (antagonist.len < 4)
```

Comment:
```suggestion
	while (length(antagonist) < 4)
```
ditto for other cases, no reason to use `.len`

Reply:
[affc886](https://github.com/goonstation/goonstation/pull/12339/commits/affc8863907b038d9f5daa6cf902819fc7f97898)